test/integration/controller/jobset_controller_test.go

```diff
@@ -349,9 +367,96 @@ var _ = ginkgo.Describe("JobSet controller", func() {
 				},
 			},
 		}),
+		ginkgo.Entry("suspended jobset will not start", &testCase{
+			makeJobSet: func(ns *corev1.Namespace) *testing.JobSetWrapper {
+				return testJobSet(ns).
+					Suspend(true)
+			},
+			updates: []*jobUpdate{
+				{
+					expectedJobSetCondition: jobset.JobSetSuspended,
+					checkJobSetState: func(js *jobset.JobSet) {
+						ginkgo.By("checking all jobs are suspended")
+						gomega.Eventually(matchJobsSuspendState, timeout, interval).WithArguments(js, true).Should(gomega.Equal(true))
+					},
+				},
+			},
+		}),
+		ginkgo.Entry("suspended jobset resumes on setting suspend to false", &testCase{
+			makeJobSet: func(ns *corev1.Namespace) *testing.JobSetWrapper {
+				return testJobSet(ns).
+					Suspend(true)
+			},
+			updates: []*jobUpdate{
+				{
+					expectedJobSetCondition: jobset.JobSetSuspended,
+					checkJobSetState: func(js *jobset.JobSet) {
+						ginkgo.By("checking all jobs are suspended")
+						gomega.Eventually(matchJobsSuspendState, timeout, interval).WithArguments(js, true).Should(gomega.Equal(true))
+					},
+				},
+				{
+					jobSetUpdateFn: func(js *jobset.JobSet) {
+						jobSetSuspend(js, false)
+					},
+					checkJobSetState: func(js *jobset.JobSet) {
+						ginkgo.By("checking all jobs are resumed")
+						gomega.Eventually(matchJobsSuspendState, timeout, interval).WithArguments(js, false).Should(gomega.Equal(true))
+					},
+				},
+				{
+					jobUpdateFn:             completeAllJobs,
+					expectedJobSetCondition: jobset.JobSetCompleted,
+				},
+			},
+		}),
+		ginkgo.Entry("jobset that gets suspended after startup transitions to suspended", &testCase{
+			makeJobSet: func(ns *corev1.Namespace) *testing.JobSetWrapper {
+				return testJobSet(ns).Suspend(false)
+			},
+			updates: []*jobUpdate{
+				{
+					checkJobSetState: func(js *jobset.JobSet) {
+						ginkgo.By("checking all jobs are not suspended")
+						gomega.Eventually(matchJobsSuspendState, timeout, interval).WithArguments(js, false).Should(gomega.Equal(true))
+					},
+				},
+				{
+					jobSetUpdateFn: func(js *jobset.JobSet) {
+						jobSetSuspend(js, true)
+					},
+					expectedJobSetCondition: jobset.JobSetSuspended,
+					checkJobSetState: func(js *jobset.JobSet) {
+						ginkgo.By("checking all jobs are suspended")
+						gomega.Eventually(matchJobsSuspendState, timeout, interval).WithArguments(js, true).Should(gomega.Equal(true))
+					},
+				},
+			},
+		}),
 	) // end of DescribeTable
 }) // end of Describe
 
+func checkJobSetStatus(js *jobset.JobSet, condition jobset.JobSetConditionType) (bool, error) {
```

Comment:
These 2 functions (`checkJobSetStatus`, `numExpectedJobs`) are now part of a shared `sigs.k8s.io/jobset/test/util` pkg so you can use those shared functions here.

Reply:
Ah thank you.  I think its rebased but it was stupid.  I'll make sure to use the util one and remove this.

Reply:
removed.